src/rules/validator.ts

```diff
@@ -66,8 +67,8 @@ export const validateConfig = (config: ConfigurationFile): ConfigurationFile | n
       validatedConfig.rules[i] = validate<BasicRule>(rule, basicRuleSchema, { message });
     } else if (type === "debug") {
       validatedConfig.rules[i] = validate<DebugRule>(rule, ruleSchema, { message });
-    } else if (type === "and") {
-      validatedConfig.rules[i] = validate<AndRule>(rule, andRuleSchema, { message });
+    } else if (type === "and" || type === "or") {
+      validatedConfig.rules[i] = validate<AndRule>(rule, otherRulesSchema, { message });
```

Comment:
Kinda feels like these rules can be joined so the validation would work in one execution

Reply:
I am afraid I don't follow. What do you mean by joined? Could you please show me an example?

I want to keep the "else" to throw if the rule is invalid, but overall, I don't worry much about removing the hardcoded code as, maybe with one or two exception, I don't think we will be adding more cases.

Reply:
I mean, you validate `config`, and then validate each rule independently.  
Can't we just join it all in a single schema to validate in one call?

Reply:
Ah yeah, I see your point.

It is because the "Basic" rule has different parameters, so I thought it would be easier to do:
```js
if (rule.type === "basic")
    basicCheck();
else if (rule.type !== "basic")
    notBasicCheck();
```

And that's is simply because *I don't know* how to make a conditional rule with JOI in which:
1. If the _type_ string has the value `basic` it should use a particular schema type 
2. if the _type_ string does not have the value `basic` it needs to have a different schema validating it.

I don't know how to do a conditional check like that. If you do know, please let me know, as I have spend some time researching that and decided to go for the simplest solution instead of fighting for the most elegant one.

I'll be merging this PR as the union should be handled in a different PR.

Reply:
You mean like `Joi.alternatives()`? 
```ts

const BasicSchema = Joi.object({
   type: "basic",  
   data: Joi.object({a: Joi.string()})
  });

const NonBasicSchema = Joi.object({
    // I'm personally more inclined to have positive checks instead
    type: Joi.any().not("basic"), 
    data: Joi.object({b: Joi.string()})
  });

Joi.alternatives().try(BasicSchema, NonBasicSchema);
```